### PR TITLE
Do not share requests session within multiple threads

### PR DESCRIPTION
**Pull request by midnightercz**
If one session object is shared within multiple threads, it can
happen it ran out of retries even for new connections which is not
desired. Constant number of retries should be applied to every
request. In this commit session object is created per every
worker thread

Fixes #7149.

### CODE REVIEW

**Comment by midnightercz** [comment]
@dkliban @daviddavis please review

File: nectar/downloaders/threaded.py [hunk 1]
```diff
@@ -175,7 +174,8 @@ def worker(self, queue):
                 request = queue.get()
                 if request is None or self.is_canceled:
                     break
-                self._fetch(request)
+                session = self._make_session()
```
Comment (ipanova) [inline nectar/downloaders/threaded.py:177]: makse_session does not return anything so `session` will be None

File: nectar/downloaders/threaded.py [hunk 2]
```diff
@@ -175,7 +174,8 @@ def worker(self, queue):
...
+                self._fetch(session, request)
```
Comment (ipanova) [inline nectar/downloaders/threaded.py:178]: please fix the order `self._fetch request, session)`

File: nectar/downloaders/threaded.py [hunk 3]
```diff
@@ -242,7 +242,7 @@ def _download_one(self, request):
         """
         return self._fetch(request)
```
Comment (ipanova) [inline nectar/downloaders/threaded.py:243]: this also needs a change

File: nectar/downloaders/threaded.py [hunk 4]
```diff
@@ -242,7 +242,7 @@ def _download_one(self, request):
...
 
-    def _fetch(self, request):
+    def _fetch(self, request, session):
         """
         :param request: download request object with details about what to
```
Comment (ipanova) [inline nectar/downloaders/threaded.py:247]: please also add `session` to the doc block

**Pull request merged**